pr/game.py

```diff
@@ -1,150 +1,184 @@
+            answer = input('New game? (Y/N) ')
+            if answer != 'Y' and answer != 'y':
+                next_game = False
+
+        self.log.log_to_file('Game is over!')
+
+    def __round_game(self):
+        self.log.log_to_file('New round!')
+        next_round = True
+
+        while next_round:
+            self.__play_round()
+            answer = input('New round? (Y/N) ')
+            if answer != 'Y' and answer != 'y':
+                next_round = False
+            else:
+                self.__prepare_round()
+
+        self.log.log_to_file('Round is over!')
+
+    def __play_round(self):
+        is_round_over = False
+
+        while not is_round_over:
+            char = self.__get_next_char()
+            column, row = self.__next_step()
+            self.__set_cell_value(char, column, row)
+            is_round_over = self.__is_round_over(char, column, row)
+            self.log.show_board(self.__board)
+
+    def __prepare_game(self):
+        self.__set_players_name()
+        self.__board_size = self.__get_value('Input board size more than 2: ', 2)
+        self.__prepare_round()
+
+    def __prepare_round(self):
+        self.__board = []
+        self.__board = self.__create_board(self.__board_size)
+        self.__cell_counter = self.__board_size * self.__board_size
+        self.__is_first_player = True
+
+    def __set_players_name(self):
+        print('Hi! Let\'s play!')
+        self.__players['x'] = input('Input first player name (it will play with x): ')
+        self.__players['o'] = input('Input second player name (it will play with o): ')
+
+    @staticmethod
+    def __get_value(message, down_board, up_board=20):
+        is_incorrect_input = True
+        player_input = 0
+        while is_incorrect_input:
+            try:
+                player_input = int(input(message))
+            except ValueError:
+                print('Incorrect input!')
+            else:
+                if player_input < down_board or player_input > up_board:
+                    print('Incorrect value!')
+                else:
+                    is_incorrect_input = False
 
+        return player_input
 
-def get_next_char():
-    global is_first_player, next_char
-    if is_first_player:
-        next_char = 'x'
-    else:
-        next_char = 'o'
-    is_first_player = not is_first_player
-    return next_char
+    def __get_next_char(self):
+        if self.__is_first_player:
+            next_char = 'x'
+        else:
+            next_char = 'o'
+        self.__is_first_player = not self.__is_first_player
+        self.__next_char = next_char
+        return next_char
+
+    def __create_board(self, size=3):
+        for i in range(size):
+            self.__board.append([' '] * size)
+        return self.__board
+
+    def __next_step(self):
+        print('Step of ' + self.__players[self.__next_char])
+        row = self.__get_value('Enter row position from 1 till ' + str(len(self.__board)) + ': ', 1,
```

Comment:
Are you sure `str` is required here?

Reply:
I agree that it's look strange. But I get TypeError without str
![err1](https://user-images.githubusercontent.com/60970966/160615933-6445a969-9a15-4706-aedf-9490c14861e8.png)

Reply:
Thanks! Realy cool suggestion! Fixed